src/OgAccess.php

```diff
@@ -112,18 +111,24 @@ public static function userAccess(EntityInterface $group, $operation, AccountInt
     // To reduce the number of SQL queries, we cache the user's permissions
     // in a static variable.
     if (!$pre_alter_cache) {
-      $permissions = array();
-
-      foreach (Og::getUserMemberships($user) as $membership) {
+      $permissions = [];
+      $user_is_group_admin = FALSE;
 
+      if ($membership = Og::getUserMembership($user, $group)) {
```

Comment:
@amitaibu this is also a pretty critical fix, before this was adding permissions for a users group from all groups the user is a part of. uh oh. Test coverage also included below. I verified locally by first writing the tests, which failed. This then fixed them again.

We are caching this stuff per group/per user so the permissions cached based on this need to reflect that too.